automate.sh

```diff
@@ -1,45 +1,45 @@
 #!/bin/bash
 
-# create the Templates directory in the home folder, if it does not exist
 if [ -f Templates.tar.gz ];
 then
 	# extract the dummy files
-	echo "Extracting.."
+	echo "Extracting files..."
 	tar -xvzf Templates.tar.gz
 
-	# check if ~/Templates already exists
-	if [ ! -d ~/Templates ];
-	then
-		mv Templates ~/Templates
-		echo -e "\nTemplates directory created.."
-	else
-		# directory exists; merge, instad
-		rsync -a Templates/ ~/Templates/
-		rm -rf Templates
+	# if .config/user-dirs.dirs does not exist, create it and append config settings to tell nautilus about our templates and
+	# create the custom Templates directory in the home folder, if it does not exist
+	if [[ ! -f ~/.config/user-dirs.dirs ]];
+		then
+			xdg-user-dirs-update
+			echo "Config file created..."
+		else
+			# this config already exists; append the entry
+			source ~/.config/user-dirs.dirs
+			customPath=$XDG_TEMPLATES_DIR
+			echo -e "\nYour Templates path for your language is" $customPath
+			
+				# check if the custom path already exists
+				if [ -d "$customPath" ];
+				then
+					# directory exists; merge
+					echo -e "Custom path already exists. Merging files...\n(If you already do have a file with the same file name in that target directory, you'll be prompted to choose which one to keep)"
+					mv -i Templates/* $customPath
+					rm -rf Templates
+				else
+					# directory doesn't exist; creating and moving files
+					mkdir $customPath
+					mv Templates/* $customPath
+					rm -rf Templates
+					echo "Created custom Templates folder and added files."
+				fi
 	fi
-
 else
 	echo -e "\nTemplates.tar.gz not found.\nPlease ensure you have the compressed tarball from: https://github.com/angela-d/nautilus-right-click-new-file in the same directory as this script.\nExiting."
 	exit 0
 fi
-echo "Context menu options created.."
-
-# if .config/user-dirs.dirs does not exist, create it and append config settings to tell nautilus about our templates
-if [[ ! -f ~/.config/user-dirs.dirs ]];
-	then
-		xdg-user-dirs-update
-		echo "Config file created.."
-	else
 
-		# this config already exists; append the entry
-		source ~/.config/user-dirs.dirs
-		if [[ $XDG_TEMPLATES_DIR = "$HOME/Templates" ]]
```

Comment:
Do a check like this! :)

Reply:
The problem I have is that I don't know all the possible translated names and that some users might also change their assigned path name manually (see my other comment above).
Also, if a new language is being supported by Gnome, the script (my pull request script) would automatically take that associated name  and we wouldn't have to pull an update script.

I don't know if there is still a way to do such a check, but as of now I cannot think of any way. If you have an idea I'd be glad to hear it from you.

As far as I know this code

> if [[ $XDG_TEMPLATES_DIR = "$HOME/Templates" ]]

looks only whether there is an entry for `$XDG_TEMPLATES_DIR` (in that case if the value is `"$HOME/Templates"`) so we would have to find a method to check whether the entry is not null.

Did I get your concern right or did you mean something else?

Reply:
I just figured it out and it was really easy. Should have been obvious ;)

Here's the file and commit: https://github.com/GithubUser699/nautilus-right-click-new-file/commit/810a9923555845db68b0080d9bbbd915e4cabdef